File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        values={
            "I":1,
            "V":5,
            "X":10,
            "L":50,
            "C":100,
            "D":500,
            "M":1000
            }
        
        N = len(s)
        i = N-1
        output = 0
        while i>=0:
            if i< N-1 and values[s[i]] < values[s[i+1]]:
                output -= values[s[i]]
            else:
                output += values[s[i]]
            i-=1
            
        return output
            
        

File: 0013-roman-to-integer/0013-roman-to-integer.js
/**
 * @param {string} s
 * @return {number}
 */

var romanToInt = function(s) {
    const values = {
    "I": 1,
    "V": 5,
    "X": 10,
    "L": 50,
    "C": 100,
    "D": 500,
    "M": 1000
}
   let res=0
   s.split('').forEach((num,i)=>{
       //if value num in front is less than next number (ex IV) subtract

       if(values[num] < values[s[i+1]]){
        res -= values[num]
        }
       else res += values[num]
   })
    
    return res
};